examples/bbox_annotator.py

```diff
@@ -35,22 +39,25 @@ def create_label_menu(shapes_layer, label_property, labels):
 
     def update_label_menu(event):
         """This is a callback function that updates the label menu when
-        the current properties of the Shapes layer change
+        the default features of the Shapes layer change
         """
-        new_label = str(shapes_layer.current_properties[label_property][0])
+        new_label = str(shapes_layer.feature_defaults[label_property][0])
         if new_label != label_menu.value:
             label_menu.value = new_label
 
-    shapes_layer.events.current_properties.connect(update_label_menu)
+    shapes_layer.events.feature_defaults.connect(update_label_menu)
 
     def label_changed(event):
-        """This is a callback that update the current properties on the Shapes layer
+        """This is a callback that update the default features on the Shapes layer
         when the label menu selection changes
         """
         selected_label = event.value
-        current_properties = shapes_layer.current_properties
-        current_properties[label_property] = np.asarray([selected_label])
-        shapes_layer.current_properties = current_properties
+        shapes_layer.feature_defaults[label_property] = selected_label
```

Comment:
@jni : I'm curious what you think of this approach to mutating feature defaults.

At first I added the `feature_defaults` setter (as in this PR) and used it successfully, but then I though about this alternative and kinda like it. What do you think?

Reply:
This was discussed in team meeting a bit - I can see pros and cons. I'm ok with it, it has a comment - though I might add a more explicit comment about the need to emit the event as it is the only place currently required. The main reason I like this approach is that if we ever get to a world where we don't need to manually emit the event it is very easy to migrate people (just get them to drop extra line). I don't have strong opinions though, so if @jni or @tlambert03 have strong views I will defer

Reply:
I'm fine with it.  No strong opinions either.  wish it weren't necessary, but it is! :)